samples/contrib/aws-samples/debugger_component_demo/debugger-component-demo.py

```diff
@@ -0,0 +1,151 @@
+
+sagemaker_train_op = components.load_component_from_file(components_dir + '/train/component.yaml')
+
+debug_hook = {
+    'S3OutputPath':'s3://kubeflow-pipeline-data/mnist_kmeans_example/hook_config'
+}
+
+debug_hook['CollectionConfigurations'] = []
+
+collection_list = {
+    'feature_importance' : {
+        'save_interval': '5'
+    }, 
+    'losses' : {
+        'save_interval': '10'
+    },
+    'average_shap': {
+        'save_interval': '5'
+    },
+    'metrics': {
+        'save_interval': '5'
+    }
+}
+
+for key, val in collection_list.items():
+    debug_hook['CollectionConfigurations'].append({'CollectionName': key, 'CollectionParameters': val})
+
+loss_rule = {
+    'RuleConfigurationName': 'LossNotDecreasing',
+    'RuleEvaluatorImage': '503895931360.dkr.ecr.us-east-1.amazonaws.com/sagemaker-debugger-rules:latest',
+    'RuleParameters': {
+        'rule_to_invoke': 'LossNotDecreasing',
+        'tensor_regex': '.*'
+    }
+}
+
+overtraining_rule = {
+    'RuleConfigurationName': 'Overtraining',
+    'RuleEvaluatorImage': '503895931360.dkr.ecr.us-east-1.amazonaws.com/sagemaker-debugger-rules:latest',
+    'RuleParameters': {
+        'rule_to_invoke': 'Overtraining',
+        'patience_train': '10',
+        'patience_validation': '20'
+    }
+}
+
+debug_rule_configurations=[loss_rule, overtraining_rule]
+
+bad_hyperparameters = {
+    'objective': 'reg:squarederror',
+    'max_depth': '5', 
+    'eta': '0', 
+    'gamma': '4', 
+    'min_child_weight': '6', 
+    'silent': '0', 
+    'subsample': '0.7', 
+    'num_round': '50'
+}
+
+channelObjList = []
+
+channelObj = {
+    'ChannelName': '',
+    'DataSource': {
+        'S3DataSource': {
+            'S3Uri': '',
+            'S3DataType': 'S3Prefix',
+            'S3DataDistributionType': 'FullyReplicated'
+        }
+    },
+    'ContentType': 'text/csv',
+    'CompressionType': 'None',
+    'RecordWrapperType': 'None',
+    'InputMode': 'File'
+}
+
+channelObj['ChannelName'] = 'train'
+channelObj['DataSource']['S3DataSource']['S3Uri'] = 's3://kubeflow-pipeline-data/mnist_kmeans_example/input/valid_data.csv'
+channelObjList.append(copy.deepcopy(channelObj))
+
+
+@dsl.pipeline(
+    name='xgboost-mnist-debugger',
+    description='SageMaker training job test with debugger'
+)
+def training(
+        region='us-east-1',
+        endpoint_url='',
+        image='683313688378.dkr.ecr.us-east-1.amazonaws.com/sagemaker-xgboost:0.90-2-cpu-py3',
+        training_input_mode='File',
+        hyperparameters=bad_hyperparameters,
+        channels=channelObjList,
+        instance_type='ml.m5.2xlarge',
+        instance_count=1,
+        volume_size=50,
+        max_run_time=3600,
+        model_artifact_path='s3://kubeflow-pipeline-data/mnist_kmeans_example/output/model',
+        output_encryption_key='',
+        network_isolation=True,
+        traffic_encryption=False,
+        spot_instance=False,
```

Comment:
Can we remove any pipeline inputs that are acting as the defaults? `spot_instance` has a default of False, so for sake of simplicity it should be removed. This is inline with [a previous PR's](https://github.com/kubeflow/pipelines/pull/4192) goals.